Zend/zend_exceptions.c

```diff
@@ -396,101 +396,177 @@ ZEND_METHOD(error_exception, __construct)
 {
-	zval *prop, rv;
-
+	zval *code;
 	ZEND_PARSE_PARAMETERS_NONE();
 
-	prop = GET_PROPERTY(ZEND_THIS, ZEND_STR_CODE);
-	ZVAL_DEREF(prop);
-	ZVAL_COPY(return_value, prop);
+	code = zend_exception_get_code(Z_OBJ_P(ZEND_THIS), 0);
+	ZVAL_COPY(return_value, code);
 }
 /* }}} */
 
+ZEND_API HashTable *zend_exception_get_trace(zend_object *exception, zend_bool silent)
+{ /* {{{ */
+	return _get_arr_prop(exception, ZEND_STR_TRACE, silent);
+} /* }}} */
+
 /* {{{ proto array Exception|Error::getTrace()
    Get the stack trace for the location in which the exception occurred */
 ZEND_METHOD(exception, getTrace)
 {
-	zval *prop, rv;
-
+	HashTable *trace;
 	ZEND_PARSE_PARAMETERS_NONE();
 
-	prop = GET_PROPERTY(ZEND_THIS, ZEND_STR_TRACE);
-	ZVAL_DEREF(prop);
-	ZVAL_COPY(return_value, prop);
+	if ((trace = zend_exception_get_trace(Z_OBJ_P(ZEND_THIS), 0))) {
+		GC_ADDREF(trace);
+		ZVAL_ARR(return_value, trace);
+	}
 }
 /* }}} */
 
 /* {{{ proto int ErrorException::getSeverity()
    Get the exception severity */
 ZEND_METHOD(error_exception, getSeverity)
 {
-	zval *prop, rv;
-
+	zend_long *sev;
 	ZEND_PARSE_PARAMETERS_NONE();
 
-	prop = GET_PROPERTY(ZEND_THIS, ZEND_STR_SEVERITY);
-	ZVAL_DEREF(prop);
-	ZVAL_COPY(return_value, prop);
+	if ((sev = _get_long_prop(Z_OBJ_P(ZEND_THIS), ZEND_STR_SEVERITY, 0))) {
+		RETURN_LONG(*sev);
+	}
 }
 /* }}} */
 
-#define TRACE_APPEND_KEY(key) do {                                          \
-		tmp = zend_hash_find(ht, key);                                      \
-		if (tmp) {                                                          \
-			if (Z_TYPE_P(tmp) != IS_STRING) {                               \
-				zend_error(E_WARNING, "Value for %s is no string",          \
-					ZSTR_VAL(key));                                         \
-				smart_str_appends(str, "[unknown]");                        \
-			} else {                                                        \
-				smart_str_appends(str, Z_STRVAL_P(tmp));                    \
-			}                                                               \
-		} \
+#define TRACE_APPEND_KEY(key, silent)                                          \
+	do {                                                                   \
+		tmp = zend_hash_find(ht, key);                                 \
+		if (tmp) {                                                     \
+			if (Z_TYPE_P(tmp) != IS_STRING) {                      \
+				if (!silent) {                                 \
+					zend_error(                            \
+						E_WARNING,                     \
+						"Trace %s is not a string",    \
+						ZSTR_VAL(key));                \
+				}                                              \
+				smart_str_appends(str, "[unknown]");           \
+			} else {                                               \
+				smart_str_appends(str, Z_STRVAL_P(tmp));       \
+			}                                                      \
+		}                                                              \
```

Comment:
Fix the indentation of the `\` on the end of each line of that macro.

Reply:
It *is* fixed. It's just that some of our code base uses 4 spaces for "tabs" and some uses 8.